lib/spack/spack/util/gcs.py

```diff
@@ -0,0 +1,216 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+"""
+This file contains the definition of the GCS Blob storage Class used to
+integrate GCS Blob storage with spack buildcache.
+"""
+
+import os
+import sys
+
+import llnl.util.tty as tty
+
+
+def gcs_client():
+    """Create a GCS client
+    Creates an authenticated GCS client to access GCS buckets and blobs
+    """
+
+    try:
+        import google.auth
+        from google.cloud import storage
+    except ImportError as ex:
+        tty.error('{0}, google-cloud-storage python module is missing.'.format(ex) +
+                  ' Please install to use the gs:// backend.')
+        sys.exit(1)
+
+    storage_credentials, storage_project = google.auth.default()
+    storage_client = storage.Client(storage_project,
+                                    storage_credentials)
+    return storage_client
+
+
+class GCSBucket(object):
+    """GCS Bucket Object
+    Create a wrapper object for a GCS Bucket. Provides methods to wrap spack
+    related tasks, such as destroy.
+    """
+    def __init__(self, url, client=None):
+        """Constructor for GCSBucket objects
+        """
+        if url.scheme != 'gs':
+            raise ValueError('Can not create GCS bucket connection with scheme {SCHEME}'
+                             .format(SCHEME=url.scheme))
+        self. url = url
+        self.name = self.url.netloc
+        if self.url.path[0] == '/':
+            self.prefix = self.url.path[1:]
+        else:
+            self.prefix = self.url.path
+
+        if not client:
+            self.client = gcs_client()
+        else:
+            self.client = client
+
+        self.bucket = None
+        tty.debug('New GCS bucket:')
+        tty.debug("    name: {0}".format(self.name))
+        tty.debug("    prefix: {0}".format(self.prefix))
+
+    def exists(self):
+        from google.cloud.exceptions import NotFound
+        if not self.bucket:
+            try:
+                self.bucket = self.client.bucket(self.name)
+            except NotFound as ex:
+                tty.error("{0}, Failed check for bucket existence".format(ex))
+                sys.exit(1)
+        return (self.bucket is not None)
```

Comment:
Minor, but parenthesis are not needed (and usually not used in Python):
```suggestion
        return self.bucket is not None
```

Reply:
👍🏻